src/components/Admin/QuestionDisplay.js

```diff
@@ -0,0 +1,212 @@
+  }
+`;
+
+class QuestionDisplay extends Component {
+  constructor(props) {
+    super(props);
+
+    this.toggleEdit = this.toggleEdit.bind(this);
+    this.toggleDelete = this.toggleDelete.bind(this);
+    this.updateData = this.updateData.bind(this);
+  }
+
+  _initFirebase = false;
+  state = {
+    loading: true,
+    error: null,
+    question: null,
+    editQuestion: false,
+    deleteQuestion: false,
+  };
+  unsub = null;
+
+  componentDidMount() {
+    if (this.props.firebase && !this._initFirebase) this.loadSettings();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.firebase && !this._initFirebase) this.loadSettings();
+  }
+
+  componentWillUnmount() {
+    if (typeof this.unsub === "function") this.unsub();
+  }
+
+  updatePage = () => {
+    this.loadSettings();
+  };
+
+  loadSettings = async () => {
+    this._initFirebase = true;
+
+    this.setState({ question: this.props.question, loading: false });
+  };
+
+  toggleEdit = () => {
+    this.setState({ editQuestion: !this.state.editQuestion });
+  };
+
+  toggleDelete = () => {
+    this.setState({ deleteQuestion: !this.state.deleteQuestion });
+  };
+
+  deleteQ = () => {
+    this.props.firebase
+      .question(this.props.question.id)
+      .delete()
+      .then(() => {
+        console.log("Question Deleted: ", this.props.question.id);
+        this.updateData();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  };
+
+  updateData = () => {
+    this.setState({ deleteQuestion: false, editQuestion: false });
+    this.props.updateFunc();
+  };
+
+  render() {
+    const {
+      loading,
+      error,
+      editQuestion,
+      question,
+      deleteQuestion,
+    } = this.state;
+
+    if (loading) return <Loader />;
+    if (error)
+      return (
+        <Container flexdirection="column">
+          <h1>{error}</h1>
+        </Container>
+      );
+
+    var hasIMG = false;
+    if (question.image !== "") hasIMG = true;
+
+    if (editQuestion) {
+      return (
+        <StyledCol md={4}>
+          <StyledDiv>
+            <h2>Edit Question</h2>
+
+            <StyledHr />
+            <EditQuestion
+              question={question}
+              uid={this.props.question.id}
+              updateFunc={this.updateData}
+            />
+            <StyledHr />
+
+            <Button onClick={this.toggleEdit}>Edit Data</Button>
+            <StyledHr />
+          </StyledDiv>
+        </StyledCol>
+      );
+    }
+
+    const Levels = () => {
+      return (
+        <Row>
+          <Col>
+            <h3> Level Scores </h3>
+
+            {Object.entries(question.scores).map((level) => (
+              <p key={level[0]}>
+                {" "}
+                {level[0]} | Average: {level[1].avrg} | Amount:{" "}
+                {level[1].amount}{" "}
+              </p>
+            ))}
+          </Col>
+        </Row>
+      );
+    };
+
+    const Delete = () => {
```

Comment:
Same here, don't need to define a component if only used once. In general, only add new components/functions/etc if the logic is used multiple times.

Reply:
Resolved in commit 21706d9246057ef3da00d313a3a1ea132dedb6c1